### ISSUE TITLE
After .ent export with script, .ent will not open anymore

### ISSUE THREAD

**Issue by Muad0510** [WolvenKit/WolvenKit#1433]
**Describe the bug**

Updated from 8.11.0 to 8.11.1, tried exporting two custom entities.
After exporting, blender would not import them from the json.
Wkit now refuses to open one of the two ents. Other opens.

**To Reproduce**
1. Have Metro rE3worked mod [Metro rE3worked 0.2.2.zip](https://github.com/WolvenKit/WolvenKit/files/13367963/Metro.rE3worked.0.2.2.zip)
2. Be on Wkit 8.11.1
3. Export .ent with built in script
4. try opening json in blender with plugin 1.5.1
5. See error in blender
6. try opening .ent in Wkit, see errors

**Expected behavior**

Not this. Open the .ent as usual.

**Version (please complete the following information):**
 - Wkit 8.11.1
 - Blender plugin 1.5.1

**Screenshots**

**Ent export:**
![image](https://github.com/WolvenKit/WolvenKit/assets/9463454/0d780223-6c95-4b51-b469-e824c54b6554)

**Error after exporting with script:**

`[15/11/2023 19:17:13] [Important] base\vehicles\special\custom_coach_e3.ent
[15/11/2023 19:17:16] [Error    ] base\gameplay\ai\archetypes\vehicles\base_vehicle.aiarch could not be found
[15/11/2023 19:17:16] [Error    ] base\fx\vehicles\_lights\veh_brake_lights.effect could not be found
[15/11/2023 19:17:18] [Error    ] test\level_design\ld_kit\textures\ruler_checker_5m_road.xbm could not be found
[15/11/2023 19:17:18] [Error    ] test\level_design\ld_kit\textures\ruler_checker_1m_wall.xbm could not be found
[15/11/2023 19:17:18] [Important] base\vehicles\special\custom_coach_new.ent
[15/11/2023 19:17:18] [Error    ] Message: Exception of type 'System.Exception' was thrown.
Source: WolvenKit.RED4
StackTrace:    at WolvenKit.RED4.IO.Red4Reader.GetStringValue(UInt16 index, Boolean isTypeInfo) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Types\IO\Red4Reader.cs:line 60
   at WolvenKit.RED4.Types.worldCompiledEffectInfo.CustomRead(Red4Reader reader, UInt32 size, CName propertyName) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Types\ClassesExt\CustomData\worldCompiledEffectInfo.cs:line 117
   at WolvenKit.RED4.Archive.IO.RedPackageReader.ReadClass(RedBaseClass cls, UInt32 size) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\RedPackageReader.cs:line 156
   at WolvenKit.RED4.IO.Red4Reader.ReadClass(List`1 redTypeInfos, UInt32 size) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Types\IO\Red4Reader.cs:line 122
   at WolvenKit.RED4.Archive.IO.RedPackageReader.ReadClass(RedBaseClass cls, UInt32 size) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\RedPackageReader.cs:line 156
   at WolvenKit.RED4.Archive.IO.RedPackageReader.ReadBuffer(RedBuffer buffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\RedPackageReader.File.cs:line 129
   at WolvenKit.RED4.Archive.IO.entEntityTemplateReader.ReadBuffer(RedBuffer buffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\entEntityTemplateReader.cs:line 18
   at WolvenKit.RED4.Archive.IO.CR2WReader.ParseBuffer(RedBuffer buffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\CR2WReader.File.cs:line 304
   at WolvenKit.RED4.Archive.IO.CR2WReader.ReadFile(CR2WFile& file, Boolean parseBuffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\CR2WReader.File.cs:line 62
   at WolvenKit.RED4.CR2W.Red4ParserService.TryReadRed4File(Stream stream, CR2WFile& redFile) in D:\a\WolvenKit\WolvenKit\WolvenKit.Common\RED4\Red4ParserService.cs:line 85
[15/11/2023 19:17:18] [Error    ] "base\vehicles\special\custom_coach_new.ent" is not a CR2W file
[15/11/2023 19:17:18] [Error    ] TypeError: Cannot read properties of null (reading 'Data')
    at ParseFile (Module:178:35)
    at Module:43:5
[15/11/2023 19:17:18] [Important] Execution time: 00:00:04.5317448
`

**Error when trying to open 'custom_coach_new.ent':**

`[15/11/2023 19:18:12] [Error    ] Message: Exception of type 'System.Exception' was thrown.
Source: WolvenKit.RED4
StackTrace:    at WolvenKit.RED4.IO.Red4Reader.GetStringValue(UInt16 index, Boolean isTypeInfo) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Types\IO\Red4Reader.cs:line 60
   at WolvenKit.RED4.Types.worldCompiledEffectInfo.CustomRead(Red4Reader reader, UInt32 size, CName propertyName) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Types\ClassesExt\CustomData\worldCompiledEffectInfo.cs:line 117
   at WolvenKit.RED4.Archive.IO.RedPackageReader.ReadClass(RedBaseClass cls, UInt32 size) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\RedPackageReader.cs:line 156
   at WolvenKit.RED4.IO.Red4Reader.ReadClass(List`1 redTypeInfos, UInt32 size) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Types\IO\Red4Reader.cs:line 122
   at WolvenKit.RED4.Archive.IO.RedPackageReader.ReadClass(RedBaseClass cls, UInt32 size) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\RedPackageReader.cs:line 156
   at WolvenKit.RED4.Archive.IO.RedPackageReader.ReadBuffer(RedBuffer buffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\RedPackageReader.File.cs:line 16
   at WolvenKit.RED4.Archive.IO.entEntityTemplateReader.ReadBuffer(RedBuffer buffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\entEntityTemplateReader.cs:line 18
   at WolvenKit.RED4.Archive.IO.CR2WReader.ParseBuffer(RedBuffer buffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\CR2WReader.File.cs:line 304
   at WolvenKit.RED4.Archive.IO.CR2WReader.ReadFile(CR2WFile& file, Boolean parseBuffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\CR2WReader.File.cs:line 62
   at WolvenKit.RED4.CR2W.Red4ParserService.TryReadRed4File(Stream stream, CR2WFile& redFile) in D:\a\WolvenKit\WolvenKit\WolvenKit.Common\RED4\Red4ParserService.cs:line 85
[15/11/2023 19:18:12] [Error    ] Message: Exception of type 'System.Exception' was thrown.
Source: WolvenKit.RED4
StackTrace:    at WolvenKit.RED4.IO.Red4Reader.GetStringValue(UInt16 index, Boolean isTypeInfo) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Types\IO\Red4Reader.cs:line 60
   at WolvenKit.RED4.Types.worldCompiledEffectInfo.CustomRead(Red4Reader reader, UInt32 size, CName propertyName) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Types\ClassesExt\CustomData\worldCompiledEffectInfo.cs:line 117
   at WolvenKit.RED4.Archive.IO.RedPackageReader.ReadClass(RedBaseClass cls, UInt32 size) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\RedPackageReader.cs:line 156
   at WolvenKit.RED4.IO.Red4Reader.ReadClass(List`1 redTypeInfos, UInt32 size) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Types\IO\Red4Reader.cs:line 122
   at WolvenKit.RED4.Archive.IO.RedPackageReader.ReadClass(RedBaseClass cls, UInt32 size) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\RedPackageReader.cs:line 156
   at WolvenKit.RED4.Archive.IO.RedPackageReader.ReadBuffer(RedBuffer buffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\RedPackageReader.File.cs:line 16
   at WolvenKit.RED4.Archive.IO.entEntityTemplateReader.ReadBuffer(RedBuffer buffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\entEntityTemplateReader.cs:line 18
   at WolvenKit.RED4.Archive.IO.CR2WReader.ParseBuffer(RedBuffer buffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\CR2WReader.File.cs:line 304
   at WolvenKit.RED4.Archive.IO.CR2WReader.ReadFile(CR2WFile& file, Boolean parseBuffer) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\IO\CR2WReader.File.cs:line 62
   at WolvenKit.RED4.CR2W.Red4ParserService.TryReadRed4File(BinaryReader br, Boolean parseBuffers, CR2WFile& redFile) in D:\a\WolvenKit\WolvenKit\WolvenKit.Common\RED4\Red4ParserService.cs:line 112
[15/11/2023 19:18:12] [Error    ] Failed to read cr2w file E:\WolvenKit\Projects\Metro rE3worked 0.2.2\source\archive\base\vehicles\special\custom_coach_new.ent
[15/11/2023 19:18:12] [Warning  ] Unable to open file: E:\WolvenKit\Projects\Metro rE3worked 0.2.2\source\archive\base\vehicles\special\custom_coach_new.ent
`

**Comment by Simarilius-uk** [comment]
if I just download the reworked open with the asset browser and try to open the custom_coach_new it doesnt like it. doesnt appear to be anything to do with the script

**Comment by Muad0510** [comment]
Earlier version of the packaged mod, which should not be broken.
[Metro rE3worked 0.2.2 beta.zip](https://github.com/WolvenKit/WolvenKit/files/13368086/Metro.rE3worked.0.2.2.beta.zip)

**Comment by DoctorPresto** [comment]
Is the .ent you're using from pre 2.0?

**Comment by Muad0510** [comment]
Yes it is. 
Before updating to Wkit 8.11.1, it did not cause any visible issues. It may have output error or warning lines, but nothing that stopped me from working with it.
Reverting back to an older version of the .ent still works, and I was able to add the new components to it (interior light, lightChannel, etc.) that the newer one had before it broke. 
Right now it works and even imports into blender with the script. After I removed the two redundant appearances that kept it as the bus instead of the metro cart.
